pr_insight/git_providers/bitbucket_server_provider.py

```
@@ -480,3 +517,36 @@ def _get_pr_comments_path(self):
+            get_logger().error("No bearer token provided. Returning None")
+            return None
+        # Return unmodified URL as the token is passed via HTTP headers in _clone_inner, as seen below.
+        return repo_url_to_clone
+
+    # Overriding the shell command, since for some reason usage of x-token-auth doesn't work, as mentioned here:
+    # https://stackoverflow.com/questions/56760396/cloning-bitbucket-server-repo-with-access-tokens
+    def _clone_inner(self, repo_url: str, dest_folder: str, operation_timeout_in_seconds: int = None):
+        bearer_token = self.bearer_token
+        if not bearer_token:
+            # Shouldn't happen since this is checked in _prepare_clone, therefore - throwing an exception.
+            raise RuntimeError(f"Bearer token is required!")
```

Ruff (0.8.2)

f-string without any placeholders

(F541)